plugin.js

```diff
@@ -90,13 +89,21 @@ function compare (a, b) {
   }
 }
 
-function plugin (fastify, options, next) {
-  fastify.addHook('onRequest', factory(options))
-  next()
+function plugin (fastify, options, done) {
+  options = Object.assign({ addHook: true }, options)
+
+  fastify.decorate('verifyBearerAuthFactory', verifyBearerAuthFactory)
+  fastify.decorate('verifyBearerAuth', verifyBearerAuthFactory(options))
```

Comment:
If `addHook` is `true`, should we still add these decorations?

Reply:
You are right, if the `hook` is added. There are no way people needing those decorations as they cannot override the verification.

Reply:
Always adding the decorators is better for the typescript definitions